Allow authentication using cached JSON data

fbis251 opened the pull request.

ccrama commented:
Is this the file from my JRAW fork? I was gonna put this in one big PR but if you'd like to that's cool :)

fbis251 commented:
Hey, I cherry picked the commit you made with regard to caching authentication data. I changed a few things to that method after some testing, then I also added another method that allows you to use cached JSON data doing userless and app-only authentication.

Although your commit is included in the pull request there are also some more changes in the request building and documentation

ccrama commented:
Gotcha, looks good!

thatJavaNerd commented on src/main/java/net/dean/jraw/http/oauth/OAuthHelper.java:
```diff
@@ -2,40 +2,42 @@
+ * 'installed' or 'web', a typical use of this class is as follows:
+ * <p>
  * <ol>
- *     <li>Obtain an authorization URL using {@link #getAuthorizationUrl(Credentials, boolean, String...)}.
```
Please revert documentation formatting

fbis251 commented:
Hey @thatJavaNerd is the last commit enough or should I revert the changes to spacing in the javadoc parameters and descriptions?

thatJavaNerd commented:
Please revert the formatting before I merge the pull request.

fbis251 commented:
I have submitted a new commit with the changes you requested

thatJavaNerd commented:
Very nice, thank you both!
